doc/sphinx/reference/onedim/grid-refinement.md

```diff
@@ -0,0 +1,229 @@
+initial guess for each subsequent refined solution lies within the domain of
+convergence of the Newton solver.
+
+The algorithm tracks information about which grid points
+should be kept and which ones need to have additional grid points inserted to the
+right of them. This data is used to refine the grid as well as to keep too many
+points from being removed during a refinement step.
+
+
+## Refinement Criteria
+
+There are 4 refinement criteria that can be set by a user to control the grid
+refinement process:
+
+  - ratio
+  - slope
+  - curve
+  - prune
+
+### Ratio
+
+This refinement option controls the maximum ratio of the spacing between the grid
+points that are adjacent to a point under consideration.If the ratio between two
+adjacent grid points is greater than the ratio
+specified by the user ($ \alpha_{max} $), then an additional grid point is inserted
+between the two points. Depending on which interval adjacent to a point under
+consideration is larger, two conditions are checked:
+
+#### Larger Left Interval
+
+For grid points that have a larger spacing on the left interval, as shown in the figure below,
+
+```{figure} /_static/images/onedim/refinement_large_left_interval.svg
+:width: 75%
+:alt: Grid with a spacing on the right interval that is too large.
+:align: center
+
+Grid with a spacing on the right interval that is too large.
+```
+
+the ratio, $ \alpha $ is calculated as:
+
+$$
+  \alpha = \frac{{z_j - z_{j-1}}}{{z_{j+1} - z_j}}
+$$
+
+#### Larger Right Interval
+
+For grid points that have a larger spacing on the right interval, as shown in the figure below,
+
+```{figure} /_static/images/onedim/refinement_large_right_interval.svg
+:width: 75%
+:alt: Grid with a spacing on the left interval that is too large.
+:align: center
+
+Grid with a spacing on the left interval that is too large.
+```
+
+the ratio, $ \alpha $ is calculated as:
+
+$$
+  \alpha = \frac{{z_{j+1} - z_j}}{{z_j - z_{j-1}}}
+$$
+
+
+These conditions are checked for all grid points in the domain, and if the ratio
+is greater than the user-specified value, then an additional grid point is inserted
+between the two points. To illustrate the interplay between the insertion of points
+in the domain and the marking of points to be kept, consider again the two cases above.
+
+For the case where the spacing on the right interval is too large, the grid is refined
+and a stencil of points around the point under consideration are marked to be kept.
+The figure below shows an example of the stencil of points that are marked to be kept
+when the spacing on the right interval is too large.
+
+```{figure} /_static/images/onedim/refinement_right_interval_stencil.svg
+:width: 75%
+:alt: The points that are marked as needing new grid points inserted to the right of
+them and which ones should be marked to be kept.
+:align: center
+
+The points that are marked as needing new grid points inserted to the right of them
+and which ones should be marked to be kept.
+```
```

Comment:
I don't know what if any syntax is needed for the alt text to extend across multiple lines, but this doesn't work. Probably the easiest fix is that there really isn't a need for alt text if it's just a repeat of the caption.